src/go/k8s/pkg/resources/certmanager/schemaregistry_api.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2021 Vectorized, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+// Package certmanager contains resources for TLS certificate handling using cert-manager
+package certmanager
+
+import (
+	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	"github.com/vectorizedio/redpanda/src/go/k8s/pkg/resources"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	schemaRegistryAPI           = "schema-registry"
+	schemaRegistryAPIClientCert = "schema-registry-client"
+	schemaRegistryAPINodeCert   = "schema-registry-node"
+)
+
+// SchemaRegistryAPINodeCert returns the namespaced name for the SchemaRegistry API certificate used by nodes
+func (r *PkiReconciler) SchemaRegistryAPINodeCert() types.NamespacedName {
+	return types.NamespacedName{Name: r.pandaCluster.Name + "-" + schemaRegistryAPINodeCert, Namespace: r.pandaCluster.Namespace}
+}
+
+// SchemaRegistryAPIClientCert returns the namespaced name for the SchemaRegistry API certificate used by clients
+func (r *PkiReconciler) SchemaRegistryAPIClientCert() types.NamespacedName {
+	return types.NamespacedName{Name: r.pandaCluster.Name + "-" + schemaRegistryAPIClientCert, Namespace: r.pandaCluster.Namespace}
+}
+
+func (r *PkiReconciler) prepareSchemaRegistryAPI(
+	issuerRef *cmmeta.ObjectReference, keystoreSecret *types.NamespacedName,
+) []resources.Resource {
+	toApply := []resources.Resource{}
+
+	// Redpanda cluster certificate for SchemaRegistry API - to be provided to each broker
+	cn := NewCommonName(r.pandaCluster.Name, schemaRegistryAPINodeCert)
+	certsKey := types.NamespacedName{Name: string(cn), Namespace: r.pandaCluster.Namespace}
+
+	dnsName := []string{r.clusterFQDN}
+	if r.pandaCluster.Spec.Configuration.SchemaRegistry != nil &&
+		r.pandaCluster.Spec.Configuration.SchemaRegistry.External != nil &&
+		r.pandaCluster.Spec.Configuration.SchemaRegistry.External.Subdomain != "" {
+		dnsName = append(dnsName, r.pandaCluster.Spec.Configuration.SchemaRegistry.External.Subdomain)
```

Comment:
so for schema registry we have tls on both internal and external ports?

Reply:
If user enable schema registry with external connectivity then the CA subject alternative name will look like:
```
openssl x509 -in ca.crt -text -noout
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            8a:37:c9:a8:63:e6:1e:d8:4d:34:4c:f3:76:25:b0:2e
    Signature Algorithm: sha256WithRSAEncryption
        Issuer: CN=rp-896b265388d62b2bd1672c0d503c-schema-registry-root-certificate
        Validity
            Not Before: Jul 23 23:31:57 2021 GMT
            Not After : Jul 22 23:31:57 2026 GMT
        Subject: CN=rp-896b265388d62b2bd1672c0d503c-schema-registry-root-certificate
       ...
        X509v3 extensions:
            X509v3 Key Usage: critical
                Digital Signature, Key Encipherment, Certificate Sign
            X509v3 Basic Constraints: critical
                CA:TRUE
            X509v3 Subject Key Identifier:
                3F:AD:6D:6B:DC:60:29:0C:17:29:45:50:57:78:23:14:73:65:01:AA
            X509v3 Subject Alternative Name:
                DNS:rp-896b265388d62b2bd1672c0d503c5c0f3a299452-cluster.rp-2-276-464.svc.cluster.local, DNS:*.rp-896b265388d62b2bd1672c0d503c5c0f3a299452-cluster.rp-2-276-464.svc.cluster.local, DNS:rp-896b265388d62b2bd1672c0d503c5c0f3a299452.59d91d3.byoc.vectorized.cloud, DNS:*.rp-896b265388d62b2bd1672c0d503c5c0f3a299452.59d91d3.byoc.vectorized.cloud
```